### PR TITLE
Standardizes on repoclosure through obal and drops Candlepin repoclos…

### PR DESCRIPTION
**Pull request by ehelms**
…ures

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.20.groovy [hunk 1]
```diff
@@ -24,20 +23,14 @@ pipeline {
...
                         parallel(
-                            'client/el7': { repoclosure('foreman-client', 'el7') },
-                            'client/el6': { repoclosure('foreman-client', 'el6') },
-                            'client/el5': { repoclosure('foreman-client', 'el5') },
-                            'client/fc28': { repoclosure('foreman-client', 'f28') },
-                            'client/fc27': { repoclosure('foreman-client', 'f27') }
+                            'client/el7': { repoclosure('foreman-client', 'el7', env.version) },
+                            'client/el6': { repoclosure('foreman-client', 'el6', env.version) },
+                            'client/el5': { repoclosure('foreman-client', 'el5', env.version) },
+                            'client/fc28': { repoclosure('foreman-client', 'f29', env.version) },
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.20.groovy]: 1.20 does not support f29 :)

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy [hunk 2]
```diff
@@ -24,20 +23,14 @@ pipeline {
...
                         parallel(
-                            'client/el7': { repoclosure('foreman-client', 'el7') },
-                            'client/el6': { repoclosure('foreman-client', 'el6') },
-                            'client/el5': { repoclosure('foreman-client', 'el5') },
-                            'client/fc28': { repoclosure('foreman-client', 'f28') },
-                            'client/fc27': { repoclosure('foreman-client', 'f27') }
+                            'client/el7': { repoclosure('foreman-client', 'el7', env.version) },
+                            'client/el6': { repoclosure('foreman-client', 'el6', env.version) },
+                            'client/el5': { repoclosure('foreman-client', 'el5', env.version) },
+                            'client/fc28': { repoclosure('foreman-client', 'f29', env.version) },
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy]: nope 28 != 29

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.10.groovy [hunk 3]
```diff
@@ -1,6 +1,10 @@
 pipeline {
     agent none
 
+    environment {
+        version = "1.20"
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.10.groovy]: should this be `foreman_version` or `packaging_version`? as Katellos version is 3.10 here (applies to all katello files)

**Comment by ehelms** [comment]
Addressed comments

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.20.groovy [hunk 4]
```diff
@@ -18,26 +17,20 @@ pipeline {
...
                         parallel(
-                            'client/el7': { repoclosure('foreman-client', 'el7') },
-                            'client/el6': { repoclosure('foreman-client', 'el6') },
-                            'client/el5': { repoclosure('foreman-client', 'el5') },
-                            'client/fc28': { repoclosure('foreman-client', 'f28') },
-                            'client/fc27': { repoclosure('foreman-client', 'f27') }
+                            'client/el7': { repoclosure('foreman-client', 'el7', env.foreman_version) },
+                            'client/el6': { repoclosure('foreman-client', 'el6', env.foreman_version) },
+                            'client/el5': { repoclosure('foreman-client', 'el5', env.foreman_version) },
+                            'client/fc28': { repoclosure('foreman-client', 'f29', env.foreman_version) },
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.20.groovy]:
```suggestion
                            'client/fc28': { repoclosure('foreman-client', 'f28', env.foreman_version) },
```

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.20.groovy [hunk 5]
```diff
@@ -18,26 +17,20 @@ pipeline {
...
+                            'client/fc27': { repoclosure('foreman-client', 'f28', env.foreman_version) }
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.20.groovy]:
```suggestion
                            'client/fc27': { repoclosure('foreman-client', 'f27', env.foreman_version) }
```

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy [hunk 6]
```diff
@@ -18,26 +17,20 @@ pipeline {
...
                         parallel(
-                            'client/el7': { repoclosure('foreman-client', 'el7') },
-                            'client/el6': { repoclosure('foreman-client', 'el6') },
-                            'client/el5': { repoclosure('foreman-client', 'el5') },
-                            'client/fc28': { repoclosure('foreman-client', 'f28') },
-                            'client/fc27': { repoclosure('foreman-client', 'f27') }
+                            'client/el7': { repoclosure('foreman-client', 'el7', env.foreman_foreman_version) },
+                            'client/el6': { repoclosure('foreman-client', 'el6', env.foreman_foreman_version) },
+                            'client/el5': { repoclosure('foreman-client', 'el5', env.foreman_foreman_version) },
+                            'client/fc28': { repoclosure('foreman-client', 'f29', env.foreman_foreman_version) },
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy]:
```suggestion
                            'client/fc28': { repoclosure('foreman-client', 'f28', env.foreman_foreman_version) },
```

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy [hunk 7]
```diff
@@ -18,26 +17,20 @@ pipeline {
...
+                            'client/fc27': { repoclosure('foreman-client', 'f28', env.foreman_foreman_version) }
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy]:
```suggestion
                            'client/fc27': { repoclosure('foreman-client', 'f27', env.foreman_foreman_version) }
```

**Comment by ehelms** [comment]
Thanks for pointing those out again. I've updated those changes.

**Comment by ehelms** [comment]
Tests were failing randomly. Fixed now.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy [hunk 8]
```diff
@@ -2,8 +2,7 @@ pipeline {
     agent none
 
     environment {
-        version = "1.21"
-        branch = "rpm/${version}"
+        foreman_foreman_version = "1.21"
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy]: why the double foreman?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-rpm.groovy [hunk 9]
```diff
@@ -1,6 +1,10 @@
 pipeline {
     agent none
 
+    environment {
+        foreman_foreman_version = 'develop'
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-rpm.groovy]: why the double foreman?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy (hunk 8 above)
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/client/1.21.groovy]: https://www.youtube.com/watch?v=hft1oYm_G2c

**Comment by tbrisker** [comment]
hopefully fix the few that are missing: https://github.com/theforeman/foreman-infra/pull/1070

**Comment by tbrisker** [comment]
I think this also broke the foreman-client repos: e.g. https://ci.theforeman.org/blue/organizations/jenkins/foreman-client-1.20-rpm-pipeline/detail/foreman-client-1.20-rpm-pipeline/78/pipeline

**Pull request merged**